hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceWALReader.java

```diff
@@ -248,7 +248,7 @@ private void handleEmptyWALEntryBatch() throws InterruptedException {
   // enabled, then dump the log
   private void handleEofException(IOException e) {
     if ((e instanceof EOFException || e.getCause() instanceof EOFException) &&
-      logQueue.size() > 1 && this.eofAutoRecovery) {
+      (source.isRecovered() || logQueue.size() > 1) && this.eofAutoRecovery) {
```

Comment:
nice find, mind adding a quick comment since this is a subtle behavior?

Reply:
On a related note, should we add a source and global metric counter to track the number of 0 size files dequeued? Seems more common than we thought, easy to monitor may be..

Reply:
> On a related note, should we add a source and global metric counter to track the number of 0 size files dequeued? Seems more common than we thought, easy to monitor may be..

@bharathv  Good idea. Could you please create a separate jira to track this. Thank you !